scripts/linear_bandit.py

```diff
@@ -0,0 +1,155 @@
+import enum
+
+import jax.numpy as jnp
+from jax import lax
+from jax import random
+
+from tensorflow_probability.substrates import jax as tfp
+
+tfd = tfp.distributions
+
+
+class ExplorationPolicy(enum.Enum):
+    """Possible exploration policies."""
+
+    linear_ucb_policy = 1
+    linear_thompson_sampling_policy = 2
+    linear_epsilon_greedy_policy = 3
+
+
+class LinearBandit:
+    def __init__(
+        self,
+        num_features,
+        num_arms,
+        exploration_policy,
+        eta=6.0,
+        lmbda=0.25,
+        alpha=1.0,
+        epsilon=0,
+    ):
+        self.num_features = num_features
+        self.num_arms = num_arms
+        self.eta = eta
+        self.lmbda = lmbda
+        self._alpha = alpha
+        self._epsilon = epsilon
+
+        if exploration_policy == ExplorationPolicy.linear_ucb_policy:
+            self._policy = self.ucb_policy
+        elif exploration_policy == ExplorationPolicy.linear_thompson_sampling_policy:
+            self._policy = self.thompson_sampling_policy
+        elif exploration_policy == ExplorationPolicy.linear_epsilon_greedy_policy:
+            self._policy = self.epsilon_greedy_policy
+        else:
+            raise NotImplemented
+
+    def _get_epsilon(self):
+        if callable(self._epsilon):
+            return self._epsilon()
+        else:
+            return self._epsilon
+
+    def init_bel(self, contexts, actions, rewards):
+        mu = jnp.zeros((self.num_arms, self.num_features))
+        Sigma = (
+            1.0
+            / self.lmbda
+            * jnp.eye(self.num_features)
+            * jnp.ones((self.num_arms, 1, 1))
+        )
```

Comment:
We could rewrite this as follows:

```
Sigma = jnp.eye(self.num_features) * jnp.ones((self.num_arms, 1, 1) / self.lmbda
```

Or even better:
```
Sigma = jnp.ones((self.num_arms, self.num_features, self.num_features)) / self.lmbda
```

Reply:
I like the last one. Thanks

Reply:
You're totally right! Thanks for pointing that out @posgnu.